torch/distributed/distributed_c10d.py

```diff
@@ -1352,8 +1352,11 @@ def _new_process_group_helper(
         pg._register_backend(torch.device(device), backend_type, backend_class)
 
     # update global state
+    assert group_name is not None
```

Comment:
OK, if we do require `group_name`, shall we remove the default `None` in L1194?

Reply:
We actually can, I searched enough for reasons to avoid this fix but there are none.